sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,942 @@
+   this SIP transitions into _Recommended_ status, so as to provide sufficient time for
+   node operators to upgrade. Tentatively this block number would be chosen to fall
+   on November 29th or November 30th, 2021.  In this document, this is the
+**activation block**.
+
+   2.  In the two whole reward cycles prior to the activation block, users who
+       have Stacked STX will have the opportunity to cast a vote to activate
+this SIP.  The cut-off for voting will be a _separate_ Bitcoin block whose
+expected arrival time is one calendar week prior to the activation block.  This
+document refers to this block as the **vote deadline block.**
+
+   3.  If the activation voting threshold is met as of the vote deadline block,
+then the Stacks Foundation will cut a relase of the Stacks blockchain
+reference implementation with this
+SIP's changes applied and set to take effect once the activation block passes.
+If on the other hand there is insufficient support for this SIP by the vote
+deadline block, then no action will be taken and this SIP will not activate.
+
+To activate this SIP, users who have Stacked STX in either of the last two whole
+reward cycles prior to the vote deadline block height will have the opportunity to
+vote with their STX by sending a minimial amount of BTC to one of two addresses.
+There will be two Bitcoin addresses whose UTXOs will be used to tally the
+vote: a "yes" address, and a "no" address.
+
+* The "yes" address will be a p2pkh Bitcoin address whose inner Hash160 is
+  `00000000000000000000000000000000000000ee`.  On mainnet, this is address
+`111111111111111111112czxoHN`.
+
+* The "no" address will be a separate p2pkh Bitcoin address whose inner Hash160
+  is `00000000000000000000000000000000000000ff`.  On mainnet, this is address
+`111111111111111111112kmzDG2`.
+
+Note that these are similar addresses to the PoX burn address, but they all
+differ from one another in their checksums.
+
+Vote tallying is performed by examining how many STX the sender has recently
+Stacked.  The Bitcoin transaction identifies the amount of STX in the `.pox`
+smart contract's data space.  So, by examining the UTXOs for these two Bitcoin
+addresses, anyone with a full copy of the Stacks chain state as of the voting
+deadline will be able to calculate how many recently-Stacked STX have signaled
+"yes" or "no" support for this SIP.
+
+To match the Bitcoin transaction to the Stacker's state in the `.pox` contract,
+the `scriptSig` of the first transaction input must be signed by either the user's PoX reward
+address's public key(s), or the public key(s) of the Stacks address that Stacked
+the tokens (the option is provided here because not all Stackers have access to their PoX
+addresses). In either case, the vote commits the Stacker's
+most-recently-locked STX to "yes" or "no" if the Stacker had some STX locked
+in the past two reward cycles as of the vote deadline block.
```

Comment:
> @friedger What does this mean for stackers that delegated their stacking?

Excellent question from @friedger over on https://github.com/stacksgov/sips/pull/43.  The answer is that in this design, the pool operator would be voting on behalf of the users that delegated to the pool.  It would be up to the pool operator to figure out which way to vote on behalf of his/her users.

I'm open to making it so that individual users could vote by sending Bitcoin transactions instead, but I'm not sure what the best way to do this would be.  A Bitcoin transaction isn't cheap, so I'm not sure it's something that users would want to do.

I'm also open to making it so that if users disagreed with the pool operator, they would have the option of de-voting their STX by revoking delegation prior to the vote deadline block.  Then, the pool's total STX voted would be the difference between the number of STX Stacked and the number of STX that are locked but no longer delegated.

Maybe we could make it so that if the user de-delegates their STX, they would have the option of sending a Bitcoin transaction to vote on their own?